cumulus/parachains/runtimes/testing/penpal/src/xcm_config.rs

```diff
@@ -318,8 +325,62 @@ pub type Reserves = (
 	NativeAssetFrom<SystemAssetHubLocation>,
 	AssetPrefixFrom<EthereumLocation, SystemAssetHubLocation>,
 );
-pub type TrustedTeleporters =
-	(AssetFromChain<LocalTeleportableToAssetHub, SystemAssetHubLocation>,);
+
+pub struct ConcreteAssetFromBridgeHub<AssetLocation>(PhantomData<AssetLocation>);
+impl<AssetLocation: Get<Location>> ContainsPair<Asset, Location>
+	for ConcreteAssetFromBridgeHub<AssetLocation>
+{
+	fn contains(asset: &Asset, origin: &Location) -> bool {
+		log::trace!(target: "xcm::contains", "ConcreteAssetFromBridgeHub asset: {:?}, origin: {:?}", asset, origin);
+		let is_from_bh = match origin.unpack() {
+			// System parachain
+			(1, [Parachain(id)]) => *id == 1013,
+			// Others
+			_ => false,
+		};
+		asset.id.0 == AssetLocation::get() && is_from_bh
+	}
+}
+
+pub type TrustedTeleporters = (
+	ConcreteAssetFromBridgeHub<RelayLocation>,
+	AssetFromChain<LocalTeleportableToAssetHub, SystemAssetHubLocation>,
+);
+
+parameter_types! {
+	pub SiblingBridgeHubLocation: Location = Location::new(
+				1,
+				[
+					Parachain(1013),
+				]
+			);
+	pub UniversalAliases: BTreeSet<(Location, Junction)> = BTreeSet::from_iter(
+			sp_std::vec![
+				(SiblingBridgeHubLocation::get(),GlobalConsensus(EthereumNetwork::get())),
```

Comment:
This needs to be the limited to the inbound-queue pallet. See https://github.com/Snowfork/polkadot-sdk/blob/snowbridge/cumulus/parachains/runtimes/assets/asset-hub-rococo/src/xcm_config.rs#L876.

Reply:
There is some difference from the previous `register_token` call on AH, as you may notice in

https://github.com/Snowfork/polkadot-sdk/blob/3ed7260d6d3fe91921e931c785cb2e11bf201292/bridges/snowbridge/primitives/router/src/inbound/mod.rs#L332-L349

It's not require to first descend to inbound queue before the transact call, instead we descend to msg.sender directly.

Reply:
I would assume that's very specific for `register_token` on AH and may not apply to the case here. 

First we can't use `ReceiveTeleportedAsset` to integrate with 3rd chain, the recent xcm instructions are as following:

https://github.com/Snowfork/polkadot-sdk/blob/e3639185b3f75db9f73856d82846e9f759d7dfcb/bridges/snowbridge/primitives/router/src/inbound/mod.rs#L332-L361

As you may notice we descend origin to `msg.sender` directly and charge fees also from the sender which means it requires end user to prefund it's sovereign account on dest chain.

Reply:
https://github.com/Snowfork/polkadot-sdk/pull/114/commits/13217f13973b8f9b64089e3185b2ca9702169da2